tests/test_response.py

```diff
@@ -81,3 +74,25 @@ def test_response_option_mimetype_init(monkeypatch):
     assert ro.static_media_types['.js'] == 'text/javascript'
     assert ro.static_media_types['.json'] == 'application/json'
     assert ro.static_media_types['.mjs'] == 'text/javascript'
+
+
+def test_response_set_stream(resp):
+    stream = BytesIO(b'dummy content')
+    content_length = 12
+
+    resp.set_stream(stream, content_length)
+
+    assert resp.stream == stream
+
+    assert resp._headers['content-length'] == str(content_length)
```

Comment:
It might be better to test via the public property, [`resp.headers`](https://falcon.readthedocs.io/en/stable/api/request_and_response_wsgi.html#falcon.Response.headers), instead of relying on the private interface.